### PR TITLE
Removes light from airlocks with an unrestricted side, makes the overlay glow in the dark

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
- Airlocks that have unres_side set will no longer have a light source
- The overlay associated with unrestricted access is now on the ABOVE_LIGHTING_PLANE plane, making it glow in the dark.
~~- Replaced an `if` chain with math.~~

## Why It's Good For The Game
When I originally made the unrestricted access feature, the overlay was meant to be something like a floor light, and I added a light source to the airlock so that the floor light wouldn't be hidden in the shadows if the airlock was otherwise in the dark.

That was many years ago and I know how to make glow-in-the-dark overlays now. As an aside, this helps nightmares not be blocked from shadow-step by a not-obvious light source.

~~The old method also had some repeated code due to wall of `if` statements. I have replaced them with math.~~

## Changelog

:cl:
balance: Airlocks with an unrestricted side no longer function as a light source. The indicator will now be visible even in the dark.
/:cl:

image

### CODE REVIEW

**Comment by Wallemations** [comment]
Couldn't you achieve the same thing with emissives rather than putting it above the lighting plane?

**Comment by YakumoChen** [comment]
The screenshot / PR convienently leaves out that the access / bolt lights are also a light source, which changes nothing for nightmares...

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -534,21 +529,13 @@
...
-		if(unres_sides & WEST)
-			var/image/I = image(icon='icons/obj/doors/airlocks/station/overlays.dmi', icon_state="unres_w")
-			I.pixel_x = -32
+		for(var/heading in list(NORTH,SOUTH,EAST,WEST))
+			if(!(unres_sides & heading))
+				continue
+			var/image/I = image(icon='icons/obj/doors/airlocks/station/overlays.dmi', icon_state="unres_[heading]")
+			I.plane = ABOVE_LIGHTING_PLANE
+			I.pixel_y = round((75.1085 * sin(57.29577 * ((0.959476 * heading) + 2.09713))) + 25.6243) //32 if [heading] is NORTH, -32 if SOUTH, and 0 for the rest
+			I.pixel_x = round(41.7909 - (73.8279 * sin(57.29577 * ((1.93816 * heading) + 4.94674)))) //32 if [heading] is EAST, -32 if WEST, and 0 for the rest
```
Comment (Mothblocks) [inline code/game/machinery/doors/airlock.dm:538]: what. the. hell

Comment (Mothblocks) [reply code/game/machinery/doors/airlock.dm:538]: what on earth do any of these numbers mean zxaber. why

Comment (Mothblocks) [reply code/game/machinery/doors/airlock.dm:538]: don't try to be clever like this, it's completely unreadable and required you have a silly comment

you can fix all of this and remove the comment by just:

```dm
switch (heading)
    if (NORTH)
        I.pixel_x = 0
        I.pixel_y = 32
    if (SOUTH)
        I.pixel_x = 0
        I.pixel_y = -32
    if (EAST)
        I.pixel_x = 32
        I.pixel_y = 0
    if (WEST)
        I.pixel_x = -32
        I.pixel_y = 0
```

Comment (zxaber) [reply code/game/machinery/doors/airlock.dm]: I just want you to know that using trigonometry to avoid a switch statement was very funny at 4AM.

**Comment by zxaber** [comment]
I changed it to use emissives rather than setting the plane. This causes the light to act more as a layer blend effect to things below it rather than a basic sprite.

image

It arguably doesn't have the same visual clarity, since it more heavily depends on what the contents of the adjacent floor are. But I quite like the effect.

By the by;
>The screenshot / PR convienently leaves out that the access / bolt lights are also a light source, which changes nothing for nightmares...

Bolt/access lights are not actually light sources, and nightmares have no issues shadow-stepping nearby, in front of, or through bolted doors with no external light source. Bolt lights glow in the dark, but that doesn't affect anything except their visuals in dark rooms.

**Comment by Wallemations** [comment]
If you end up not liking the effect you could simply go back to the overlay you had before but also add an emissive on top

**Comment by zxaber** [comment]
I'd like to see it in play as it now is. If I get a lot of feedback about visual clarity being poor, I can make a follow up to fix it.

**Comment by Wallemations** [comment]
> I'd like to see it in play as it now is. If I get a lot of feedback about visual clarity being poor, I can make a follow up to fix it.

Well my main concern with this is, have you checked how well it shows up in a well-lit room?
Because if you're ONLY using the emissives, then from what I understand it won't show up in well lit areas at all.

**Comment by zxaber** [comment]
When i checked it earlier, I could see it, but then I was also looking for it. I'll test it again in a bit and maybe get a screenshot.

You might be right though.

**Comment by JohnFulpWillard** [comment]
Labeling this as a Fix because of the existence of Nightmares, and the context that we recently added unrestricted helpers to most of maintenance.

**Comment by JohnFulpWillard** [comment]
The lights don't seem to be fully see-in-the-dark
image
image

**Comment by zxaber** [comment]
Yeah, as it turns out, emissives make it *really* hard to see in bright light. Such is life.

I changed them back to overlays.

**Pull request merged**